src/notification.rs

```diff
@@ -38,70 +41,66 @@ pub struct MustacheData {
     sites: Vec<Url>,
 }
 
-impl MustacheData {
-    pub fn new(commit: Option<String>, changed_urls: Vec<Url>) -> Self {
-        let mut sites = changed_urls;
-        sites.sort_unstable();
-        sites.dedup();
-
-        let mut hosts = sites
-            .iter()
-            .filter_map(Url::host_str)
-            .map(std::string::ToString::to_string)
-            .collect::<Vec<_>>();
-        hosts.dedup();
-
-        let singlehost = if let [single] = hosts.as_slice() {
+impl From<Summary> for MustacheData {
+    fn from(summary: Summary) -> Self {
+        let changed_hosts = summary.changed_hosts.into_keys().collect::<Vec<_>>();
+        let singlehost = if let [single] = changed_hosts.as_slice() {
             Some(single.clone())
         } else {
             None
         };
-
         Self {
-            commit,
+            commit: summary.commit,
             singledomain: singlehost.clone(),
             singlehost,
-            siteamount: sites.len(),
-
-            domains: hosts.clone(),
-            hosts,
-            sites,
+            siteamount: summary.changed_amount,
+            domains: changed_hosts.clone(),
+            hosts: changed_hosts,
+            sites: summary.changed_sites,
         }
     }
+}
 
+impl MustacheData {
     pub fn apply_to_template(
         &self,
         template: Option<&mustache::Template>,
-    ) -> anyhow::Result<String> {
+    ) -> Result<String, mustache::Error> {
```

Comment:
Why?

Reply:
`anyhow` is the most generic error. It's not needed to be generic, just return the exact `Result` of the inner function without wrapping it in `anyhow::Result`.